pkgs/development/python-modules/scikit-build-core/default.nix

```diff
@@ -23,62 +30,55 @@
 
 buildPythonPackage rec {
   pname = "scikit-build-core";
-  version = "0.9.6";
+  version = "0.10.5";
   pyproject = true;
 
-  src = fetchPypi {
-    pname = "scikit_build_core";
-    inherit version;
-    hash = "sha256-e+r5M89zSsvrttlsApNlQQIkcJvN5o87C08MsD4FSTk=";
+  src = fetchFromGitHub {
+    owner = "scikit-build";
+    repo = "scikit-build-core";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-hpwXEWPofgMT4ua2tZI1mtGbaBkT2XPBd6QL8xTi1A0=";
   };
 
   postPatch = lib.optionalString (pythonOlder "3.11") ''
     substituteInPlace pyproject.toml \
-      --replace '"error",' '"error", "ignore::UserWarning",'
+      --replace-fail '"error",' '"error", "ignore::UserWarning",'
   '';
 
-  nativeBuildInputs = [
+  build-system = [
     hatch-vcs
     hatchling
   ];
 
-  propagatedBuildInputs =
-    [ packaging ]
+  dontUseCmakeConfigure = true;
```

Comment:
What pulls in cmake?

Reply:
The tests do, not sure if that propagates cmake's hooks.

Reply:
Yes, the configure hook would get executed. Moving below nativeCheckInputs.